addons/crm_livechat/static/src/core/common/message_model_patch.js

```diff
@@ -0,0 +1,15 @@
+import { Message } from "@mail/core/common/message_model";
+
+import { patch } from "@web/core/utils/patch";
+
+patch(Message.prototype, {
+    get notificationHidden() {
+        if (
+            this.notificationType === "create-lead" &&
+            (this.store.self_user?.share !== false || !this.store.self_user)
```

Comment:
`|| !this.store.self_user` is not necessary then, `!== false` will catch both `true` and `undefined`